### ISSUE TITLE
Incorrect element is focused when opening mobile content switcher via keyboard

### ISSUE THREAD

**Issue by sarayourfriend** [WordPress/openverse-frontend#641]
## Description

The openverse logo button is focused when opening the mobile content switcher via keyboard instead of the content types themselves.

Marking this as high because although for sighted keyboard users who don't rely on a screen-reader this won't be too big of an issue (more of an annoyance than something breaking the usability of the site), it does cause a loss of context for users who depend on screen-readers.

## Reproduction

1. Run `main` and shrink your window to a small enough width to trigger the mobile content switcher modal.
2. Using your keyboard, tab into the content switcher button in the header and press enter to open it.
3. Note that the openverse logo is focused instead of the content type buttons.

## Expectation

When opening the content switcher, the active content button should be focused.

This may require adding back some props that were eventually removed from the original `VModal` component for passing a specific node to focus. Currently the `useDialogContent` composable just focuses the first focusable item inside the dialog rather than allowing you to pass a specific node to focus. This works for most cases but breaks in cases like this.

## Resolution

- [x] 🙋 I would be interested in resolving this bug.
